docs/topics/sealed-classes.md

```diff
@@ -1,99 +1,350 @@
 
 If a sealed class is declared as `expect` in a common source set and have `actual` implementations in platform source sets,
 both `expect` and `actual` versions can have subclasses in their source sets. Moreover, if you use a hierarchical structure,
 you can create subclasses in any source set between the `expect` and `actual` declarations. 
 
 [Learn more about the hierarchical structure of multiplatform projects](multiplatform-hierarchy.md). 
 
-## Sealed classes and when expression
+## Use sealed classes with when expression
 
 The key benefit of using sealed classes comes into play when you use them in a [`when`](control-flow.md#when-expression)
-expression. 
-If it's possible to verify that the statement covers all cases, you don't need to add an `else` clause to the statement:
+expression.
+The `when` expression, used with a sealed class, allows the Kotlin compiler to check exhaustively that all possible cases are covered. 
+In such cases, you don't need to add an `else` clause. Here's an example demonstrating this:
 
 ```kotlin
+// Sealed class and its subclasses
+sealed class Error {
+    class FileReadError(val file: String): Error()
+    class DatabaseError(val source: String): Error()
+    object RuntimeError : Error()
+}
+
+//sampleStart
+// Function to log errors
 fun log(e: Error) = when(e) {
-    is FileReadError -> { println("Error while reading file ${e.file}") }
-    is DatabaseError -> { println("Error while reading from database ${e.source}") }
-    is RuntimeError ->  { println("Runtime error") }
-    // the `else` clause is not required because all the cases are covered
+    is Error.FileReadError -> println("Error while reading file ${e.file}")
+    is Error.DatabaseError -> println("Error while reading from database ${e.source}")
+    Error.RuntimeError -> println("Runtime error")
+    // No `else` clause is required because all the cases are covered
+}
+//sampleEnd
+
+// List all errors
+fun main() {
+    val errors = listOf(
+        Error.FileReadError("example.txt"),
+        Error.DatabaseError("usersDatabase"),
+        Error.RuntimeError
+    )
+
+    errors.forEach { log(it) }
 }
 ```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.5"}
 
-> `when` expressions on [`expect`](multiplatform-expect-actual.md) sealed classes in the common code of multiplatform projects still 
-> require an `else` branch. This happens because subclasses of `actual` platform implementations aren't known in the 
-> common code.
+> In multiplatform projects, if you have a sealed class with a `when` expression as an 
+> [expected declaration](multiplatform-expect-actual.md) in your common code, you still need an `else` branch. 
+> This is because subclasses of `actual` platform implementations may extend sealed classes that 
+> aren't known in the common code.
 >
 {type="note"}
+
+## Use case scenarios
+
+Let's explore some practical scenarios where the application of sealed classes or interfaces can be particularly helpful.
```

Comment:
Let's simplify the second part of the sentence a bit:

"... where sealed classes and interfaces can be particularty helpful/useful"

Does it work?

Reply:
Yes, I think it sounds much better! - I'll update it